### PR TITLE
[Chore] Bump deps post-v1

### PR DESCRIPTION
**Pull request by justinkambic**
## Summary

Bumps dependencies for most packages. We didn't do this before v1 so as to avoid any unforeseen instability.

## Implementation details

Biggest question mark is updating `electron`, there may be some breaking change impacts as we jumped 20-24.

## How to validate this change

Make sure the recorder is working. We can do a release build to try and make sure as well.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,5 @@ eslint-junit.xml
 *-junit.xml
 .DS_Store
 tsconfig.tsbuildinfo
+dump.rdb
```
Comment (vigneshshanmugam) [inline .gitignore:16]: Ah did you try running `redis-cli` from the recorder repo path? Not sure if we need this here anyway.

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

File: package.json [hunk 2]
```diff
@@ -93,7 +93,7 @@
     "playwright": "https://gitpkg.now.sh/elastic/playwright/packages/playwright-core?1.32.3-recorder",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
-    "styled-components": "^5.3.7",
+    "styled-components": "^6.0.0-rc.1",
```
Comment (vigneshshanmugam) [inline package.json:96]: is this worth doing right now given its a release candidate? Also why not just go to rc.3 if we want to proceed with the RC.

Comment (justinkambic) [reply package.json:96]: I think I upgraded for `npm audit` reasons. Agreed though, will downgrade 👍

File: .gitignore (hunk 1 above)
Comment (justinkambic) [reply .gitignore:16]: It's possible but typically not. I'll remove it anyway as I've never needed it before, probably something weird I did.

File: package.json (hunk 2 above)
Comment (justinkambic) [reply package.json]: Fixed c7782d57f90684101fb7fa16e4f98b510fad26f6

File: .gitignore (hunk 1 above)
Comment (justinkambic) [reply .gitignore]: Fixed c7782d57f90684101fb7fa16e4f98b510fad26f6

**Pull request merged**